pkg/apis/core/v1/conversion.go

```diff
@@ -327,6 +329,35 @@ func Convert_v1_Pod_To_core_Pod(in *v1.Pod, out *core.Pod, s conversion.Scope) e
 	return nil
 }
 
+func Convert_v1_PodStatus_To_core_PodStatus(in *v1.PodStatus, out *core.PodStatus, s conversion.Scope) error {
+	autoConvert_v1_PodStatus_To_core_PodStatus(in, out, s)
+	// autoconvert sets the core PodIPs to the v1 PodIPs
+	// but does not handle the singular PodIP
+
+	if len(in.PodIPs) > 0 && in.PodIP != "" {
+		if in.PodIPs[0].IP != in.PodIP {
+			return fmt.Errorf("Invalid PodIP")
+		}
+	}
+
+	if len(out.PodIPs) > 0 {
```

Comment:
Looks like this doesn't handle the case where in.PodIPs[] is set, but in.PodIP is not set (e.g. an update operation). Maybe what we need would be something like this:
```
    // Save existing PodIPs array before autoconvert overwrites it
    savedPodIPs = out.PodIPs
    autoConvert_v1_PodStatus_To_core_PodStatus(in, out, s)

    if len(in.PodIPs) == 0 {
        if in.PodIP != "" {
            // Only PodIP is provided
            // Restore saved PodIPs
            out.PodIPs = savedPodIPs
            out.PodIPs[0] = in.PodIP
            // DELETE ANY REDUNDANT ENTRIES IN out.PodIPs[] HERE
        } else {
            // Nothing to do
            return nil
        }
    } else if in.PodIP == "" {
        // Only PodIPs[] provided
        // DELETE ANY REDUNDANT ENTRIES IN out.PodIPs[] HERE
    } else {
        // Both PodIP and PodIPs[] provided
        if in.PodIPs[0].IP != in.PodIP {
            return fmt.Errorf("Invalid PodIP")
        }
    }
```

Reply:
Yea, for this case the podIPs that is passed with the versioned, will be set in the internal version through the
autoconvert function. If the previous value is not passed in, then there are no previous values to delete.
The previous values are not passed in from what I  can see.